core/src/test/scala/tech/beshu/ror/unit/acl/blocks/rules/JwtAuthRuleTests.scala

```diff
@@ -485,4 +524,19 @@ class JwtAuthRuleTests
       .returning(Task.now(authenticated))
     service
   }
+
+  private def cachedAuthService(authenticatedToken: String, unauthenticatedToken: String) = {
+    val service = mock[ExternalAuthenticationService]
+    (service.authenticate _)
+      .expects(where { credentials: Credentials => credentials.secret === PlainTextSecret(authenticatedToken.nonempty) })
```

Comment:
I'd add here the constraint that it will be able to mock both tokens only once. Because if cache is broken, the test should fail

Reply:
scalamock has such constraint by default. I even checked it out by breaking cache and test indeed failed. Although I added `.once()` to each mock to make it explicit.